File: Daily Challanges/2026-05-May/2026-05-08.js
function medicationReminder(medications, currentTime) {
  function toMin(t) {
    const [h, m] = t.split(":").map(Number);
    return h * 60 + m;
  }

  const now = toMin(currentTime);
  let bestName = "";
  let bestWait = Infinity;

  for (const [name, last] of medications) {
    const lastMin = toMin(last);
    let nextT;

    if (name === "Mergeflictamine") {
      nextT = lastMin + 240;
      while (nextT <= now) {
        nextT += 240;
      }
    } else if (name === "Deployxitrin") {
      const schedule = [480, 960];
      nextT = schedule.find((t) => t > now);
      if (nextT === undefined) nextT = schedule[0] + 1440;
    } else {
      const schedule = [420, 780, 1260];
      nextT = schedule.find((t) => t > now);
      if (nextT === undefined) nextT = schedule[0] + 1440;
    }

    const wait = nextT - now;

    if (wait < bestWait) {
      bestWait = wait;
      bestName = name;
    }
  }

  const h = Math.floor(bestWait / 60);
  const m = bestWait % 60;

  return `${bestName} in ${h}h ${m}m`;
}

console.log(
  medicationReminder(
    [
      ["Deployxitrin", "08:00"],
      ["Debuggamanizole", "07:00"],
      ["Mergeflictamine", "10:00"],
    ],
    "11:00",
  ),
);

File: Daily Challanges/2026-05-May/2026-05-08.py
def medication_reminder(medications, current_time):
    def to_min(t):
        h, m = map(int, t.split(":"))
        return h * 60 + m

    now = to_min(current_time)
    best_name = ""
    best_wait = float("inf")

    for name, last in medications:
        last = to_min(last)

        if name == "Mergeflictamine":
            next_t = last + 240
            while next_t <= now:
                next_t += 240
        elif name == "Deployxitrin":
            schedule = [480, 960]
            next_t = next((t for t in schedule if t > now), schedule[0] + 1440)
        else:
            schedule = [420, 780, 1260]
            next_t = next((t for t in schedule if t > now), schedule[0] + 1440)

        wait = next_t - now

        if wait < best_wait:
            best_wait = wait
            best_name = name

    return f"{best_name} in {best_wait//60}h {best_wait%60}m"